client/src/amplify-ui/block-switcher/block-switcher.tsx

```diff
@@ -49,24 +59,43 @@ export class AmplifyBlockSwitcher {
     }
   }
 
+  handleInitSelectedTabHeading() {
+    if (this.tabHeadings?.length > 0 && this.selectedTabHeadings?.length > 0) {
+      for (const heading of this.selectedTabHeadings) {
+        if (this.tabHeadings.includes(heading)) {
+          this.activeChildI = this.tabHeadings.indexOf(heading);
+          return;
+        }
+      }
+    } else {
+      if (typeof window === "undefined") {
+        this.activeChildI = 0;
+      } else {
+        this.activeChildI = 0;
+      }
+    }
+  }
+
   gatherHeadings() {
     // gather tab headings from child `amplify-block` attrs
     this.tabHeadings = [];
     this.recursivelyFindBlocks(this.el);
+  }
 
-    // default to the first tab
-    this.activeChildI = 0;
+  /* eslint-disable */
+  @Event({
```

Comment:
Really surprised that this needs a lint ignore statement; I was pretty sure that both `@Watch` and `@Event` statements were fine with respect to linting -- in any case, I think we usually use `// @ts-ignore` instead!

Reply:
🤷

Apparently, the `stencil` lint plugin wants the `activeBlockUpdated: EventEmitter<string>` bit to be on the same line as the @Event decorator... but `prettier` don't like that.